packages/tuono-router/src/types.ts

```diff
@@ -5,26 +5,38 @@ export interface Segment {
   value: string
 }
 
-export interface ServerRouterInfo {
+/**
+ * Provided from the the rust server and use in ssr env
+ * @see tuono {@link ServerPayloadLocation}
+ */
+export interface ServerInitialLocation {
   href: string
   pathname: string
   searchStr: string
 }
 
-export interface ServerProps<TProps = unknown> {
-  router: ServerRouterInfo
-  props: TProps
-  jsBundles: Array<string>
-  cssBundles: Array<string>
-  mode: 'Dev' | 'Prod'
-  devServerConfig: {
-    port: number
-    host: string
-  }
-}
-
+/**
+ * @todo This could be refined using a union with isLoading: true | false.
+ *       Using this user should be able to have a better typechecking on their end:
+ *       In order to have the correct type they will be forced to add a if (isLoading) to
+ *       handle loading status.
+ * e.g.:
+ * ```ts
+ * { isLoading: false; data: TData } | { isLoading: true; data: TData }
+ * ```
+ *
+ * ```ts
+ * if (isLoading) {
+ *   // data is undefined
+ * }
+ *
+ * // data is TData
+ * ```
+ *
+ */
```

Comment:
This could be a fine addition to type checking when writing pages in tuono. 
If you agree I can open a new task and work on it in a separate PR.

Reply:
I'm not sure if there is the need. A router that does not have a relative `.rs` file does not have any loading.